Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -0,0 +1,214 @@
+    /// </example>
+    /// <example>
+    /// <code>
+    /// New-IshSession -WsBaseUrl "https://example.com/ISHWS/" -PSCredential "Admin"
+    /// $ishBackgroundTask  = Add-IshBackgroundTask -EventType "PUSHTRANSLATIONS" -EventDescription "Custom event description" -RawInputData $rawData
+    /// </code>
+    /// <para>Add background task with the event type "PUSHTRANSLATIONS" and provided event description and input data</para> 
+    /// </example>
+
+    [Cmdlet(VerbsCommon.Add, "IshBackgroundTask", SupportsShouldProcess = false)]
+    [OutputType(typeof(IshBackgroundTask))]
+    public sealed class AddIshBackgroundTask : BackgroundTaskCmdlet
+    {
+        /// <summary>
+        /// <para type="description">The IshSession variable holds the authentication and contract information. This object can be initialized using the New-IshSession cmdlet.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshSession IshSession { get; set; }
+
+        /// <summary>
+        /// <para type="description">Type of the event (e.g. SMARTTAG)</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        [ValidateNotNullOrEmpty]
+        public string EventType { get; set; }
+
+        /// <summary>
+        /// <para type="description">The input data for the background task.</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [ValidateNotNullOrEmpty]
+        public byte[] RawInputData { get; set; }
+
+        /// <summary>
+        /// <para type="description">Description of the event</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [ValidateNotNullOrEmpty]
+        public string EventDescription { get; set; }
+
+        /// <summary>
+        /// <para type="description">The <see cref="IshObjects"/>s that will be used for background task creation.</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "IshObjectsGroup")]
+        [AllowEmptyCollection]
+        public IshObject[] IshObject { get; set; }
+
+        #region Private fields
+        private readonly List<IshObject> _retrievedIshObjects = new List<IshObject>();
+        private readonly DateTime _modifiedSince = DateTime.Today.AddDays(-1);
```

Comment:
Should be start time of the cmdlet, anything earlier does not matter.

Reply:
Let's talk about this. We still find on a progressId and with modifiedSince it can be safer to look from yesterday because of  possible time difference between Client (ishRemote) and the Server ?

Reply:
Will use date of yesterday(agreed on the meeting)